js/taskFunctions.js

```diff
@@ -17,6 +17,9 @@ function printAllTasks() {
 
```

Comment:
search exist && status exist ?  do not filter at all (return true)
status exist -> search exist? yes -> filter by both : no - filter by status only

filter by search only